src/m365/spo/commands/spo-set.ts

```diff
@@ -45,6 +45,7 @@ class SpoSetCommand extends SpoCommand {
 
   public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
     auth.service.spoUrl = args.options.url;
+    auth.updateAvailableIdentitiesList();
```

Comment:
Why do we need to update the list of identities here?

Reply:
Because we want the SharePoint url to be available on the identity, also if you switch back and forth. Using this command we ensure that the spoUrl is also added to the identity object in the availableIdentities list. (Besides it being added to the active identity properties)

Reply:
Understood. In this case, I find from the name `updateAvailableIdentitiesList` totally not obvious that the method is persisting service information. We already have logic for persisting connection information and from the consumer point of view it should be opaque. I think we're leaking here too many implementation details to auth consumers. Let's look into encapsulating it inside auth so that it's done automatically whenever auth changes.

Reply:
We can change this to storeConnectionInfo() and save the array including the latest version of the active identity... that better?

Reply:
I think we've got two pieces here:

- the list of signed in accounts, which is already stored in MSAL cache, right?
- the currently active account which is stored on the service

For the latter, it makes perfect sense to have it be a part of the storeConnectionInfo logic. For the list of accounts, I wonder if we need to do something else beyond what's already available in MSAL.

Reply:
> if we need to do something else beyond what's already available in MSAL.

We do for two reasons:
1) We have some CLI-specific values that we need like authType and such. We can't get these from msal. 
2) I believe app only accounts arent' in the msal accounts list, so they aren't easily retrievable.

In short, we do need an array of availableIdentities